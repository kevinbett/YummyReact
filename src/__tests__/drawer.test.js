import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import LeftDrawer from '../components/homepage/Drawer';

describe('LeftDrawer Component', () => {
    const wrapper = shallow(<LeftDrawer />);
    it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});