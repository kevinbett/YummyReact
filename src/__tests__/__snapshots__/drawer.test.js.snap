// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftDrawer Component renders without crashing 1`] = `
<BrowserRouter>
  <div>
    <Dialog
      actions={
        Array [
          <FlatButton
            disabled={false}
            fullWidth={false}
            keyboardFocused={true}
            label="submit"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={true}
            secondary={false}
          />,
          <FlatButton
            disabled={false}
            fullWidth={false}
            keyboardFocused={false}
            label="close"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          />,
        ]
      }
      autoDetectWindowHeight={true}
      autoScrollBodyContent={false}
      modal={false}
      onRequestClose={[Function]}
      open={false}
      repositionOnUpdate={true}
      title="Add a new category"
    >
      <form>
        <TextField
          disabled={false}
          errorText=""
          floatingLabelFixed={false}
          floatingLabelText="Category name"
          fullWidth={false}
          hintText="Enter category name"
          multiLine={false}
          onChange={[Function]}
          required={true}
          rows={1}
          type="text"
          underlineShow={true}
        />
      </form>
    </Dialog>
    <Drawer
      containerStyle={
        Object {
          "overflowX": "hidden",
        }
      }
      disableSwipeToOpen={false}
      docked={true}
      open={null}
      openSecondary={false}
      swipeAreaWidth={30}
      width={null}
      zDepth={0}
    >
      <AppBar
        showMenuIconButton={false}
        style={
          Object {
            "backgroundColor": "#f57c00",
          }
        }
        title="Yummy Recipes"
        zDepth={1}
      />
      <List>
        <ListItem
          autoGenerateNestedIndicator={true}
          containerElement="span"
          disableKeyboardFocus={false}
          disabled={false}
          initiallyOpen={true}
          insetChildren={false}
          nestedItems={
            Array [
              <ListItem
                autoGenerateNestedIndicator={true}
                containerElement="span"
                disableKeyboardFocus={false}
                disabled={false}
                initiallyOpen={false}
                insetChildren={false}
                leftIcon={<ImageAddToPhotos />}
                nestedItems={Array []}
                nestedLevel={0}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onNestedListToggle={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                open={null}
                primaryText="Add category"
                primaryTogglesNestedList={false}
                secondaryTextLines={1}
              />,
              <ListItem
                autoGenerateNestedIndicator={true}
                containerElement="span"
                disableKeyboardFocus={false}
                disabled={false}
                initiallyOpen={false}
                insetChildren={false}
                leftIcon={<ImageRemoveRedEye />}
                nestedItems={Array []}
                nestedLevel={0}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onNestedListToggle={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                open={null}
                primaryText="View all categories"
                primaryTogglesNestedList={false}
                secondaryTextLines={1}
              />,
            ]
          }
          nestedLevel={0}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="Categories"
          primaryTogglesNestedList={true}
          secondaryTextLines={1}
        />
      </List>
      <Divider
        inset={false}
      />
      <List>
        <ListItem
          autoGenerateNestedIndicator={true}
          containerElement="span"
          disableKeyboardFocus={false}
          disabled={false}
          initiallyOpen={true}
          insetChildren={false}
          nestedItems={
            Array [
              <ListItem
                autoGenerateNestedIndicator={true}
                containerElement="span"
                disableKeyboardFocus={false}
                disabled={false}
                initiallyOpen={false}
                insetChildren={false}
                leftIcon={<ImageAddToPhotos />}
                nestedItems={Array []}
                nestedLevel={0}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onNestedListToggle={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                open={null}
                primaryText="Change password "
                primaryTogglesNestedList={false}
                secondaryTextLines={1}
              />,
              <ListItem
                autoGenerateNestedIndicator={true}
                containerElement="span"
                disableKeyboardFocus={false}
                disabled={false}
                initiallyOpen={false}
                insetChildren={false}
                leftIcon={<ImageRemoveRedEye />}
                nestedItems={Array []}
                nestedLevel={0}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onNestedListToggle={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                open={null}
                primaryText="Reset"
                primaryTogglesNestedList={false}
                secondaryTextLines={1}
              />,
            ]
          }
          nestedLevel={0}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="User"
          primaryTogglesNestedList={true}
          secondaryTextLines={1}
        />
      </List>
    </Drawer>
  </div>
</BrowserRouter>
`;
